vagrant/src/main/java/org/jclouds/vagrant/internal/VagrantNodeRegistry.java

```diff
@@ -80,8 +82,17 @@ public long getDelay(TimeUnit unit) {
    private Supplier<Long> timeSupplier;
 
    @Inject
-   VagrantNodeRegistry(@TimeStamp Supplier<Long> timeSupplier) {
+   VagrantNodeRegistry(@TimeStamp Supplier<Long> timeSupplier, VagrantExistingMachines existingMachines) {
       this.timeSupplier = timeSupplier;
+      this.nodes.putAll(getExistingNodes(existingMachines));
```

Comment:
It would be better to make the `VagrantExistingMachines` a supplier, to lazily load them. Otherwise all the load and parsing happens at injection time, in the constructor, where we should avoid throwing IO exceptions and other exceptions that could be raised while loading the existing machines.